drv/transceivers-server/src/udp.rs

```diff
@@ -149,64 +154,57 @@ impl ServerImpl {
-                    None
-
-                // In this case, the message has failed to deserialize, but
-                // we've successfully deserialized the header. That implies that
-                // the host has sent us a message that does not exist in our own
-                // implementation of the protocol.
-                //
-                // Check that implication by ensuring that the host version is
-                // after our own CURRENT.
-                } else if header.version > version::CURRENT {
-                    let n = hubpack::serialize(
-                        tx_data_buf,
-                        &Message {
-                            header,
-                            modules: ModuleId::default(),
-                            body: MessageBody::SpResponse(SpResponse::Error(
-                                Error::VersionMismatch {
-                                    expected: version::CURRENT,
-                                    actual: header.version,
-                                },
-                            )),
-                        },
-                    );
-                    Some(n.unwrap() as u32)
-
-                // This last case catches failures to deserialize something we
-                // _should_ have been able to deserialize. The host version is
-                // between MIN and CURRENT, so it's a message that's part of our
-                // own protocol. We really just failed to deserialize it, such
-                // as a corrupt buffer or some similar failure mode. Just drop
-                // the message, since we've already logged a `DeserializeError`.
-                } else {
-                    None
-                }
-            }
-            Err(e) => {
-                ringbuf_entry!(Trace::DeserializeHeaderError(e));
-                None
-            }
+        // This message comes from a host implementation older than the
+        // minimum committed version of the protocol. We really can't do
+        // anything with this message, and the protocol mandates that we
+        // drop the message.
+        if header.version() < version::outer::MIN {
+            ringbuf_entry!(Trace::WrongVersion(header.version()));
+            None
+
+        // In this case, the message has failed to deserialize, but
+        // we've successfully deserialized the header. That implies that
+        // the host has sent us a message that does not exist in our own
+        // implementation of the protocol.
+        //
+        // Check that implication by ensuring that the host version is
+        // after our own CURRENT.
+        } else if header.version() > version::outer::CURRENT {
+            let header_size = hubpack::serialize(
+                tx_data_buf,
+                &Header::new(header.message_id, header.message_kind),
+            )
+            .unwrap();
+            let message_size = hubpack::serialize(
+                tx_data_buf,
```

Comment:
```suggestion
                tx_data_buf[header_size..],
```
otherwise, I think we overwrite the header by accident

Reply:
Good catch!